symplyphysics/core/experimental/points/__init__.py

```diff
@@ -0,0 +1,129 @@
+"""
+Points
+======
+
+A **point** in a 3D space is described by 3 coordinates which depend on the choice of the
+coordinate system.
+"""
+
+from __future__ import annotations
+
+from typing import Sequence, Iterator, Any
+
+from sympy import Expr
+
+from ...expr_comparisons import expr_equals
+from ..coordinate_systems import BaseCoordinateSystem
+
+
+class Point:
+    """
+    Class defining the `Point` functionality.
+    """
+
+    _coordinates: tuple[Expr, Expr, Expr]
+    _system: BaseCoordinateSystem
+
+    @property
+    def coordinates(self) -> tuple[Expr, Expr, Expr]:
+        """
+        Coordinates of the point.
+        """
+
+        return self._coordinates
+
+    @property
+    def system(self) -> BaseCoordinateSystem:
+        """
+        Coordinate system the point is attached to.
+        """
+
+        return self._system
+
+    def __init__(
+        self,
+        coordinates: Sequence[Expr],
+        system: BaseCoordinateSystem,
+    ) -> None:
+        if len(coordinates) != 3:
+            raise ValueError("Only 3-dimensional points are supported.")
+
+        self._coordinates = coordinates[0], coordinates[1], coordinates[2]
+        self._system = system
+
+    def __getitem__(self, name: str) -> Expr:
+        names = type(self.system).scalar_names()
+        try:
+            index = names.index(name)
+        except ValueError as e:
+            raise IndexError(
+                f"Expected a scalar name from {names}, got {name!r}."
+            ) from e
+
+        return self.coordinates[index]
+
+    def __iter__(self) -> Iterator[Expr]:
+        return iter(self.coordinates)
+
+    def __str__(self) -> str:
+        sys_name = type(self.system).__name__
+        index = sys_name.find("CoordinateSystem")
+        name = f"{sys_name[:index]}Point"
+        return f"{name}{self.coordinates}"
```

Comment:
I regularly use the console for displaying stuff when deriving proofs, for instance, so it seems reasonable to me to add some string representation of these new classes. It's better than just writing `point.coordinates` since it also includes the type of coordinate system.